scala3doc/src/dotty/dokka/translators/ScalaPageCreator.scala

```diff
@@ -34,47 +34,37 @@ class ScalaPageCreator(
 
   override def pageForModule(m: DModule): ModulePageNode = super.pageForModule(m)
 
-  private def updatePageNameForMember(page: PageNode, p: Member) =
-    val name = page.getName
-
-    page.modified(name, page.getChildren)
-
-  private def pagesForMembers(p: Member): Seq[PageNode] =
-    p.allMembers.filter(_.origin == Origin.DefinedWithin).collect {
-      case f: DFunction => updatePageNameForMember(pageForFunction(f), f)
-      case c: DClass => updatePageNameForMember(pageForDClass(c), c)
-    }
+  private def pagesForMembers(member: Member): JList[PageNode] =
+    val (all, _) = member.membersBy(_.kind.isInstanceOf[Classlike])
+    all.map(pageForMember(_)).asJava
 
   override def pageForPackage(p: DPackage): PackagePageNode =
-    val originalPage = super.pageForPackage(p)
-    val originalPages: Seq[PageNode] = originalPage.getChildren.asScala.toList
-    val allPage: Seq[PageNode] = originalPages ++ pagesForMembers(p)
-    originalPage.modified(p.getName, allPage.asJava)
-
-  override def pageForClasslike(c: DClasslike): ClasslikePageNode = c match {
-      case clazz: DClass => pageForDClass(clazz)
-      case other => throw UnsupportedOperationException("Only DClass classlike is supported.")
-    }
-
-  def pageForDClass(c: DClass): ClasslikePageNode = {
-    val constructors = c.getConstructors
+    PackagePageNode(
+      p.name,
+      contentForPackage(p),
+      JSet(p.dri),
+      p,
+      pagesForMembers(p),
+      JNil
+    )
 
-    val ext = c.get(ClasslikeExtension)
+  override def pageForClasslike(c: DClasslike): ClasslikePageNode = ???
```

Comment:
Maybe we should add more explanatory message why calling this method fails than "Not Implemented"